Added Encryption Backup Key wizzard in post login.

anisha-e10 opened the pull request:
Implemented, https://github.com/acterglobal/a3/issues/2956

### **_Feature: Encryption Backup Key Wizard in Post-Login_**

This PR introduces a new wizard to the post-login onboarding flow, guiding users through the process of setting up an encryption backup key. This enhancement aims to bolster account security and ensure data recoverability.\

### ✅ Recovery State: Enabled

_**Step 1: Prompt to Enter Backup Key**_

- The user is shown a view where they can enter their existing encryption backup key.'
- If the user provides a valid key: they proceed securely to the next onboarding step.

**_Step 2: Continue Options_**

- A warning view appears, simulating a decryption failure scenario.

### ❌ Recovery State: Disabled

**_Step 1: Missing Encryption Key View_**

- The user is informed that the encryption key is missing.

Two options are presented:

**Try Again**: Brings the user back to the view where they can enter the encryption key.
**Continue Without Key**: Skips key entry and moves forward.

**_Step 2: Continue Options_**

- A warning view appears, simulating a decryption failure scenario.

### Reference Images : 

![Screenshot 2025-05-20 at 5 20 12 PM](https://github.com/user-attachments/assets/87a299c5-3b59-4971-8ae0-ee887ec5d971)
![Screenshot 2025-05-19 at 10 37 33 AM](https://github.com/user-attachments/assets/89aab009-bb86-438e-b6ae-b1706532ef2f)
9c0)
![Screenshot 2025-05-19 at 10 39 20 AM](https://github.com/user-attachments/assets/936c57ac-abba-409d-acfb-193eb054a28d)

### Reference Video (when the recovery state is disabled) :

https://github.com/user-attachments/assets/1addeeb8-97ff-489a-b4e9-47f4b9f6f3a7

### Reference Video (when the recovery state is enabled but incomplete) :

https://github.com/user-attachments/assets/9de1d9e7-3795-4362-afb2-a7db7df5a663

### Reference Video (when the recovery state is incomplete and recovering key is successful) :

https://github.com/user-attachments/assets/24e7bd56-0a0c-4097-9d5a-666981d1dbc0

### Reference Video (when the recovery state is incomplete and recovering key is Failure) :

https://github.com/user-attachments/assets/f3477da0-1f60-465f-b3cb-dce5e106b2e0

gnunicorn commented:
@anisha-e10 can we vertically center that text+input-box like we do with other wizard pages, please?

anisha-e10 commented:
> @anisha-e10 can we vertically center that text+input-box like we do with other wizard pages, please?

@gnunicorn I have updated UI as per other wizard pages and updated image and video in above description.

kumarpalsinh25 commented on app/lib/features/onboarding/pages/onboarding_page.dart:
```diff
@@ -68,7 +72,8 @@ class _OnboardingPageState extends ConsumerState<OnboardingPage> {
           OnboardingSpaceCreationPage(callNextPage: () => _nextPage()),
       ],
-      
+      if (!stateEnabled) 
+        OnboardingEncryptionKeyBackupPage(callNextPage: () => _nextPage()),
```
```suggestion
      if (backupState == RecoveryState.incomplete) 
        OnboardingEncryptionRecoveryPage(callNextPage: () => _nextPage()),
      if (backupState == RecoveryState.disabled) 
        MissingEncryptionBackupPage(callNextPage: () => _nextPage()),
```

kumarpalsinh25 commented on app/lib/features/onboarding/widgets/expect_decryption_failures_widget.dart:
```diff
@@ -0,0 +1,117 @@
+
+class ExpectDecryptionFailures extends ConsumerWidget {
+  final VoidCallback callNextPage;
+  final bool isRecoveryStateDisabled;
```
```suggestion
```
Doesn't need of `isRecoveryStateDisabled`

kumarpalsinh25 commented on app/lib/features/onboarding/widgets/expect_decryption_failures_widget.dart:
```diff
@@ -0,0 +1,117 @@
+  const ExpectDecryptionFailures({
+    super.key,
+    required this.callNextPage,
+    this.isRecoveryStateDisabled = false,
```
```suggestion
```

kumarpalsinh25 commented on app/lib/features/onboarding/widgets/expect_decryption_failures_widget.dart:
```diff
@@ -0,0 +1,117 @@
+        const SizedBox(height: 20),
+        OutlinedButton(
+          onPressed: () {
+            if (isRecoveryStateDisabled) {
+              Navigator.pop(context);
+            }  
```
```suggestion
        
```

kumarpalsinh25 commented on app/lib/features/onboarding/widgets/missing_encryption_backup_widget.dart:
```diff
@@ -0,0 +1,131 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:phosphor_flutter/phosphor_flutter.dart';
+
+class MissingEncryptionBackupWidget extends ConsumerWidget {
```
This should be managed as Page as explained above.
```suggestion
class MissingEncryptionBackupPage extends ConsumerWidget {
```

anisha-e10 replied:
ok will add as a page

kumarpalsinh25 commented on app/lib/features/onboarding/widgets/missing_encryption_backup_widget.dart:
```diff
@@ -0,0 +1,131 @@
+                isScrollControlled: true,
+                backgroundColor: Theme.of(context).scaffoldBackgroundColor,
+                builder: (context) {
+                  return ExpectDecryptionFailures(callNextPage: () {
+                    callNextPage();
+                  }, isRecoveryStateDisabled: true);
```
```suggestion
                  return ExpectDecryptionFailures(callNextPage: callNextPage);
```

kumarpalsinh25 reviewed (changes requested):
Routing structure based on the different state is not that Good. 

I would like to manage `OnboardingEncryptionRecoveryPage` and `MissingEncryptionBackupPage` at the main `onboarding_page.dart` level instead of putting code in `Build` method with condition.

anisha-e10 commented on app/lib/features/onboarding/pages/onboarding_encryption_key_backup_page.dart:
```diff
@@ -0,0 +1,245 @@
+    return state == RecoveryState.disabled;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // Check if recovery state is unknown and show dialog automatically
+    if (_isRecoveryStateDisabled() && !_hasShownDialog) {
+      _hasShownDialog = true;
+      WidgetsBinding.instance.addPostFrameCallback((_) {
+        showModalBottomSheet(
```
ok will do

anisha-e10 commented:
> Routing structure based on the different state is not that Good.
> 
> I would like to manage `OnboardingEncryptionRecoveryPage` and `MissingEncryptionBackupPage` at the main `onboarding_page.dart` level instead of putting code in `Build` method with condition.

Addressed by this [commit](https://github.com/acterglobal/a3/pull/2978/commits/76187a2d03d29c16fb994c760fda5c00ba8de431)
